### ISSUE TITLE
osu-lazer on cosmic-epoch alpha

### ISSUE THREAD

**Issue by Luk45135** [fufexan/nix-gaming#198]
I'm testing cosmic at the moment so this might be related, but osu-lazer just segmentation faults for me.

```
❯ osu-lazer
gamemodeauto:
gamemodeauto:
gamemodeauto:
gamemodeauto:
gamemodeauto:
gamemodeauto:
gamemodeauto:
gamemodeauto:
zsh: segmentation fault (core dumped)  osu-lazer
```
My config (on cosmic branch): https://github.com/Luk45135/nixos-config/tree/cosmic

osu-lazer logs if helpful:
[1723676081.performance.log](https://github.com/user-attachments/files/16619663/1723676081.performance.log)
[1723676081.network.log](https://github.com/user-attachments/files/16619664/1723676081.network.log)
[1723676081.auth.log](https://github.com/user-attachments/files/16619665/1723676081.auth.log)
[1723676081.runtime.log](https://github.com/user-attachments/files/16619666/1723676081.runtime.log)

**Comment by Luk45135** [comment]
feel free to disregard this issue because it works fine on hyprland and cosmic being in alpha still has lots of problems

**Comment by fufexan** [comment]
Have things improved since then when running osu on Cosmic?

**Comment by Luk45135** [comment]
no, the welcome screen opens for a split second then it crashes, sometimes it takes a bit until it crashes but i don't remember getting into the menu.

**Comment by fufexan** [comment]
Hmm, then let's keep this issue open in case others experience this. Will close when it's fixed on Cosmic.

**Comment by Luk45135** [comment]
Still doesn't work on alpha 4.

**Comment by Luk45135** [comment]
Good news! osu launches and gets into gameplay, but with horrible performance like 8fps in the menu and 16fps during gameplay.

**Comment by Luk45135** [comment]
Now i sometimes get into gameplay with normal performance but never to a result screen (after playing a map)
but very weird crashes and not consistent at all sometimes it crashes on entering daily challenges, sometimes not sometimes at entering song select sometimes when entering gameplay

and not sure if this a related issue, but i also sometimes get Bass error 46 or base driver error 3 (maybe i fixed it but i cant be sure because it's so inconsistent)

can somebody test this aswell im at a loss here.

**Comment by Luk45135** [comment]
Soo... after (what i thought was extensive testing spoiler it was not) i got fed up with it and i installed plasma6 to test it there and i had the same problem on plasma6 with these weird inconsistent freezes. Upon further testing i realized that the problem was the Vulkan renderer and switching to the default OpenGL fixes these weird freezes. I suspect that all the issues that i previously had were already fixed december last year. I apologize for keeping this issue open for so unnecessarily long, although i will open a new issue about the Vulkan renderer causing weird freezes.

tldr: problem was Vulkan. use OpenGL to avoid freezes
